src/context/box-repository.ts

```diff
@@ -149,4 +149,13 @@ export class BoxRepository extends BaseRepository<BoxEntity> {
         };
       });
   }
+
+  public async getMaxBoxIndex(): Promise<number | undefined> {
+    const { index } = await this.repository
+      .createQueryBuilder("box")
+      .select("MAX(index)", "index")
```

Comment:
Sorry I forgot to tell you, you should use `globalIndex` field in place of `index` here.

Same for transactions.